### PR TITLE
feat: add controller test generation command rough draft

### PR DESCRIPTION
**Pull request by PThorpe92**
This provides a new command: `artisan app:make-controller-test {NameOfController}`

That will prompt the user for information about the controller, model, and visitbility and generate a class of tests that will test for all the basic CRUD operations and assert that either the route is protected (will fail if non-admin makes requests) or it will assert that the user will be returned only data that belongs to the user.

### CODE REVIEW

**Comment by PThorpe92** [comment]
## The CLI will prompt the user for some information about the Controller they are trying to test
image

## and it will generate such a test class:
image
image

**Comment by PThorpe92** [comment]
This has been tested with `enrollments`.. 

upon running the command and providing the model and route, and immediately running `sail artisan test` this program will correctly generate passing tests
image

File: tests/Feature/EnrollmentControllerTest.php [hunk 1]
```diff
@@ -15,40 +16,84 @@ class EnrollmentControllerTest extends TestCase
...
+        $responseUser = $this->actingAs($user)->getJson($this->uri);
+
+        $responseAdmin = $this->actingAs($admin)->getJson($this->uri);
+
+        $responseAdmin->assertJsonCount(10, 'data');
+
+        $responseUser->assertJsonCount(5, 'data');
     }
 
-    // .... i have no words for some of these tests. We can do better
```
Comment (chrissantillan) [inline tests/Feature/EnrollmentControllerTest.php:31]: "We can do better" ... is that a specifically or a general philosophical statement haha?

**Review by chrissantillan — APPROVED** [review]

File: tests/Feature/EnrollmentControllerTest.php (hunk 1 above)
Comment (PThorpe92) [reply tests/Feature/EnrollmentControllerTest.php:31]: Lol I guess technically both? I forgot I committed that :rofl:

**Pull request merged**